pkg/controller/httpTriggerApi.go

```diff
@@ -132,7 +132,11 @@ func (a *API) checkHTTPTriggerDuplicates(t *fv1.HTTPTrigger) error {
 			// Same resource. No need to check.
 			continue
 		}
-		if ht.Spec.RelativeURL == t.Spec.RelativeURL && ht.Spec.Method == t.Spec.Method && ht.Spec.Host == t.Spec.Host {
+		urlMatch := false
+		if ht.Spec.RelativeURL == t.Spec.RelativeURL || (ht.Spec.Prefix != nil && t.Spec.Prefix != nil && *ht.Spec.Prefix != "" && *ht.Spec.Prefix == *t.Spec.Prefix) {
```

Comment:
Do we need to check if the value is an empty string? I don't think it's possible to run into that scenario because of the `omitempty` json struct tag, it is supposed to remove the field(set it as nil) if the value is empty string